From e8f06791c1ad5be3a5ac716a74b6a47c7baafaff Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 7 Mar 2018 10:16:50 +0530
Subject: [PATCH 0164/1851] dmaengine: xilinx_frmbuf: Add support for BGR8
 color format in Framebuffer

commit 247060b96139b44047ee27ce7d18931af7566175 from
https://github.com/Xilinx/linux-xlnx.git

Add support for BGR8 color format (Blue in LSB, 8 bits per component)
in Framebuffer.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_frmbuf.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/dma/xilinx/xilinx_frmbuf.c b/drivers/dma/xilinx/xilinx_frmbuf.c
index 9d98d39d22a0..6119a37e3386 100644
--- a/drivers/dma/xilinx/xilinx_frmbuf.c
+++ b/drivers/dma/xilinx/xilinx_frmbuf.c
@@ -90,6 +90,7 @@
 #define XILINX_FRMBUF_FMT_BGRA8			26
 #define XILINX_FRMBUF_FMT_BGRX8			27
 #define XILINX_FRMBUF_FMT_UYVY8			28
+#define XILINX_FRMBUF_FMT_BGR8				29
 
 /**
  * struct xilinx_frmbuf_desc_hw - Hardware Descriptor
@@ -212,7 +213,7 @@ static const struct xilinx_frmbuf_format_desc xilinx_frmbuf_formats[] = {
 		.ppw = 1,
 		.num_planes = 1,
 		.drm_fmt = DRM_FORMAT_XRGB8888,
-		.v4l2_fmt = V4L2_PIX_FMT_XRGB32,
+		.v4l2_fmt = V4L2_PIX_FMT_XBGR32,
 		.fmt_bitmask = BIT(2),
 	},
 	{
@@ -355,6 +356,16 @@ static const struct xilinx_frmbuf_format_desc xilinx_frmbuf_formats[] = {
 		.v4l2_fmt = V4L2_PIX_FMT_Y10,
 		.fmt_bitmask = BIT(17),
 	},
+	{
+		.dts_name = "rgb888",
+		.id = XILINX_FRMBUF_FMT_BGR8,
+		.bpw = 24,
+		.ppw = 1,
+		.num_planes = 1,
+		.drm_fmt = DRM_FORMAT_RGB888,
+		.v4l2_fmt = V4L2_PIX_FMT_BGR24,
+		.fmt_bitmask = BIT(18),
+	},
 };
 
 /**
-- 
2.31.1

